packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -486,125 +594,125 @@ export class RunningSummarizer implements IDisposable {
         }
 
-        const telemetryProps: any = {
-            ...summaryData,
-            ...summaryData.summaryStats,
-            refSequenceNumber: summaryData.referenceSequenceNumber,
-            opsSinceLastAttempt: summaryData.referenceSequenceNumber - this.heuristics.lastAttempted.refSequenceNumber,
-            opsSinceLastSummary: summaryData.referenceSequenceNumber - this.heuristics.lastAcked.refSequenceNumber,
-        };
-        telemetryProps.summaryStats = undefined;
-        telemetryProps.referenceSequenceNumber = undefined;
+        try {
+            const pendingTimeoutP = this.pendingAckTimer.start().catch(() => undefined);
+            const summary = this.summaryWatcher.watchSummary(summaryData.clientSequenceNumber);
 
-        if (summaryData.submitted) {
-            summarizingEvent.end(telemetryProps);
-        } else {
-            summarizingEvent.cancel(telemetryProps);
-        }
+            // Wait for broadcast
+            const summaryOp = await Promise.race([summary.waitBroadcast(), pendingTimeoutP]);
+            if (!checkNotTimeout(summaryOp)) {
+                return fail("summaryOpWaitTimeout", "Timeout while waiting for summarize op broadcast");
+            }
+            this.heuristics.lastAttempted.summarySequenceNumber = summaryOp.sequenceNumber;
+            this.logger.sendTelemetryEvent({
+                eventName: "SummaryOp",
+                timeWaiting: Date.now() - this.heuristics.lastAttempted.summaryTime,
+                refSequenceNumber: summaryOp.referenceSequenceNumber,
+                summarySequenceNumber: summaryOp.sequenceNumber,
+                handle: summaryOp.contents.handle,
+            });
 
-        return summaryData;
+            // Wait for ack/nack
+            const ackNack = await Promise.race([summary.waitAckNack(), pendingTimeoutP]);
+            if (!checkNotTimeout(ackNack)) {
+                return fail("summaryAckWaitTimeout", "Timeout while waiting for summaryAck/summaryNack op");
+            }
+            this.pendingAckTimer.clear();
+
+            // Update for success/failure
+            const telemetryProps: Record<string, number> = {
+                timeWaiting: Date.now() - this.heuristics.lastAttempted.summaryTime,
+                sequenceNumber: ackNack.sequenceNumber,
+                summarySequenceNumber: ackNack.contents.summaryProposal.summarySequenceNumber,
+            };
+            if (ackNack.type === MessageType.SummaryAck) {
+                this.heuristics.ackLastSent();
+
+                // since we need a full summary after context reload, we only clear this on ack
+                this.immediateSummary = false;
+
+                summarizeEvent.end({ ...telemetryProps, handle: ackNack.contents.handle, message: "summaryAck" });
+                return true;
+            } else {
+                return fail(
+                    "summaryNack",
```

Comment:
Note: We no longer emit "SummaryAck" or "SummaryNack" events now. Instead we send "Summarize_end" with message "summaryAck", and "Summarize_cancel" with message "summaryNack".